app/services/foreman/plugin.rb

```diff
@@ -234,15 +238,34 @@ def permission(name, hash, options = {})
 
     # Add a new role if it doesn't exist
     def role(name, permissions)
-      @default_roles[name] = permissions
+      default_roles[name] = permissions
       return false if pending_migrations || Rails.env.test?
 
       Role.transaction do
         role = Role.where(:name => name).first_or_create
```

Comment:
actually I was wrong, dupes are created, it also needs update in `add_permissions` method to skip existing permissions, this seems to do the trick

```rb
grouped_permissions.each do |permission|
  next if filter.permissions.include?(permission) # this line is required
  filtering = filter.filterings.build
  filtering.filter = filter
  filtering.permission = permission
end

```